wandb/sdk/artifacts/artifact.py

```diff
@@ -1686,6 +1686,78 @@ def download(
         root = root or self._default_root()
         self._add_download_root(root)
 
+        if wandb.run is None:
+            with wandb.init(
+                entity=self._source_entity,
+                project=self._source_project,
+                job_type="auto",
+                settings=wandb.Settings(silent="true"),
+            ):
+                return FilePathStr(
+                    self._run_artifact_download(
+                        root=root,
+                        recursive=recursive,
+                        allow_missing_references=allow_missing_references,
+                    )
+                    or ""
+                )
+        else:
+            return FilePathStr(
+                self._run_artifact_download(
+                    root=root,
+                    recursive=recursive,
+                    allow_missing_references=allow_missing_references,
+                )
+                or ""
+            )
+
+    def _run_artifact_download(
+        self,
+        root: Optional[str] = None,
+        recursive: bool = False,
+        allow_missing_references: bool = False,
+    ) -> Optional[FilePathStr]:
```

Comment:
Why is the return value optional?

Reply:
If we raise an error, then there's no return value right?

Reply:
Never mind. At first i thought since we have a path where we raise an error, this was ok. But i changed it now.